src/handler.ts

```diff
@@ -1,94 +1,807 @@
+  where,
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const query = {
+      ...where,
+      include: {
+        dag_blocks: {
+          include: {
+            global_snapshots: { select: { hash: true, ordinal: true } }
+          }
+        }
+      },
+      orderBy: { ordinal: 'desc' }
+    };
+
+    const toCursor = (row) => ({
+      ...toCreatedAtOrdinalCursor(row),
+      hash: row.hash
+    });
+
+    const fromCursor = (row) => ({
+      ...fromCreatedAtOrdinalCursor(row),
+      hash: row.hash
+    });
+
+    return await paginatedQuery(
+    extractPagination(event),
+      toCursor,
+      fromCursor,
+      query,
+      prisma.dag_transactions.findMany,
+      dagTransactionsResponse
+    );
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const transactions = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  return transactionsQuery({}, event);
+};
+
+export const transaction = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { hash } = event.pathParameters || {};
+
+    const transaction = await prisma.dag_transactions.findUnique({
+      where: { hash },
+      include: {
+        dag_blocks: {
+          include: {
+            global_snapshots: { select: { hash: true, ordinal: true } }
+          }
+        }
+      }
+    });
+
+    return respond(transaction, dagTransactionResponse);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const transactionsByAddress = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { address } = event.pathParameters || {};
+
+    const where = {
+      where: { OR: [{ source_addr: address }, { destination_addr: address }] }
+    };
+
+    return transactionsQuery(where, event);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const transactionsBySource = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { address } = event.pathParameters || {};
+
+    const where = { where: { source_addr: address } };
+
+    return transactionsQuery(where, event);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const transactionsByDestination = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { address } = event.pathParameters || {};
+
+    const where = { where: { destination_addr: address } };
+
+    return transactionsQuery(where, event);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const balanceByAddress = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { address, ordinal } = event.pathParameters || {};
+
+    const ordinalNbr = toNumber(ordinal);
+    const ordinalCondition = (isFinite(ordinalNbr)? {metagraph_snapshot_ordinal: ordinalNbr}: {})
```

Comment:
This looks incorrect to me. There shouldn't be a `metagraph_snapshot_ordinal` if we're searching for DAG balances.